src/mail/view/MailView.ts

```diff
@@ -652,27 +640,73 @@ export class MailView implements CurrentView {
 		moveMails({mailModel: locator.mailModel, mails: mailsToMove, targetMailFolder: folder})
 	}
 
-	private createFolderAddButton(mailGroupId: Id, parentFolder: MailFolder | null, folderSystem: FolderSystem): IconButtonAttrs {
+	private createFolderAddButton(mailGroupId: Id, parentFolder: MailFolder | null): IconButtonAttrs {
 		return {
 			title: "addFolder_action",
 			click: () => {
-				return this.showCreateFolderDialog(mailGroupId, parentFolder, folderSystem)
+				return this.showFolderAddEditDialog(mailGroupId, null, parentFolder)
 			},
 			icon: Icons.Add,
 			size: ButtonSize.Compact,
 		}
 	}
 
-	private showCreateFolderDialog(mailGroupId: string, parentFolder: MailFolder | null, folderSystem: FolderSystem) {
-		const infoMsg = parentFolder ? () => getPathToFolderString(folderSystem, parentFolder) : null
-		const parentFolderId: IdTuple | null = parentFolder ? parentFolder._id : null
-		Dialog.showProcessTextInputDialog("folderNameCreate_label", "folderName_label", infoMsg, "", async (name) => {
-				if (parentFolderId) {
-					this.expandedState.add(elementIdPart(parentFolderId))
+	private async showFolderAddEditDialog(mailGroupId: string, folder: MailFolder | null = null, parentFolder: MailFolder | null = null) {
+		const noParentFolderOption = "No Parent Folder";
+		const folderNameValue = stream(folder?.name ? folder?.name : "")
+		// TODO getMailboxDetails wont work for shared mailboxes?
+		const mailBoxDetail = await this.getMailboxDetails()
+		let targetFolders: SelectorItemList<MailFolder | null> = mailBoxDetail.folders.getIndentedList(folder)
+																			  .map(folderInfo => {
+																				  return {
+																					  name: getIndentedFolderNameForDropdown(folderInfo),
+																					  value: folderInfo.folder,
+																				  }
+																			  })
+		targetFolders = [{name: noParentFolderOption, value: null}, ...targetFolders]
+		let selectedParentFolder = parentFolder ? parentFolder : null
+		// @ts-ignore
+		let form = () => [
+			m(TextField, {
+				label: folder ? "rename_action" : "folderName_label",
+				value: folderNameValue(),
+				oninput: folderNameValue
+			}),
+			m(DropDownSelector, {
+				label: "parentFolder_label",
+				items: targetFolders,
+				selectedValue: selectedParentFolder,
+				selectedValueDisplay: selectedParentFolder ? getFolderName(selectedParentFolder) : noParentFolderOption,
+				selectionChangedHandler: (newFolder: MailFolder | null) => selectedParentFolder = newFolder,
+			}),
+		]
+		const okAction = (dialog: Dialog) => {
+			// TODO: make the mail facade update function
+			//locator.mailFacade.updateMailFolder(editedFolder, mailGroupId).then(() => {
+			return locator.mailFacade.createMailFolder(folderNameValue(), selectedParentFolder ? selectedParentFolder._id : null, mailGroupId).then(() => {
+				dialog.close()
+			}).catch(error => {
```

Comment:
Probably someone else should also have a look at the error handling, I'm not too familiar with that